src/org/thoughtcrime/securesms/BaseActionBarActivity.java

```diff
@@ -38,26 +38,16 @@ protected void onCreate(Bundle savedInstanceState) {
       forceOverflowMenu();
     }
     super.onCreate(savedInstanceState);
-  }
-
-    @Override
-    protected void onStart() {
-        if (ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp() && !isWaitingForResult) {
-          ScreenLockUtil.applyScreenLock(this);
-        } else if (isHiddenByScreenLock) {
-          findViewById(android.R.id.content).setVisibility(View.VISIBLE);
-          isHiddenByScreenLock = false;
-        }
-        super.onStart();
-    }
 
-  @Override
-  protected void onStop() {
-    if (ScreenLockUtil.isScreenLockEnabled(this)) {
-      findViewById(android.R.id.content).setVisibility(View.GONE);
+    if (shouldLock()) {
+      // Already hide everything here to prevent sensible data from popping up shortly:
+      findViewById(android.R.id.content).setVisibility(View.INVISIBLE);
       isHiddenByScreenLock = true;
     }
-    super.onStop();
+  }
+
+  private boolean shouldLock() {
+    return ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp() && !isWaitingForResult;
```

Comment:
This line causes the screen lock on my phone to fail and show ConversationList briefly.

The suggestion inverts the logic of "isWaitingForResult" here so that shouldLock returns true if it is already locking.

This seems to cause the lock screen to appear more then once if it failed and then got entered correctly, so this might not be the best solution.

```suggestion
    return isWaitingForResult || (ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp());
```

Reply:
Note that the suggested change below fixes the "multiple screen locks" issue, so together they worked fine on my phone.

Reply:
Actually, why check for isWaitingForResult here at all? Because, `isWaitingForResult || (ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp());` will return true if isWaitingForResult is true and getShouldLockApp() is false (i.e. the user did not even enable screen locking at all).

OTOH, currently this is not a problem because in onCreate() the screen is only blanked. Then in onActivityResult() isWaitingForResult is set to false anyway and in onResume() the screen is set to visible again.

You understand what I mean?

By the way, thanks for digging into this confusing issue with me!